src/realm/parser/query_builder.cpp

```diff
@@ -885,4 +885,82 @@ void apply_ordering(DescriptorOrdering& ordering, ConstTableRef target, const pa
 }
 
 } // namespace query_builder
+
+namespace {
+
+class MixedArguments : public Arguments {
+public:
+    MixedArguments(const std::vector<Mixed>& args)
+        : m_args(args)
+    {
+    }
+    bool bool_for_argument(size_t n)
+    {
+        return m_args[n].get<bool>();
```

Comment:
Use `m_args.at()`, because the query parser does not perform any bounds checking.